Sources_v3/ChatClient.swift

```diff
@@ -363,7 +249,7 @@ public class Client<ExtraData: ExtraDataTypes> {
     }
     
     // TODO: Not used & tested yet -> CIS-224
-    private func refreshToken(completion: @escaping (Error?) -> Void) {
+    func refreshToken(completion: @escaping (Error?) -> Void) {
```

Comment:
`refreshToken` and `prepareEnvironmentForNewUser` are the candidates for moving to **CurrentUserControllerGeneric**

At the same time, it's also IMHO OK to leave those methods in a **ChatClient** as they are mostly about modifying the data owned by a **ChatClient**

Reply:
I think `prepareEnvironmentForNewUser` can/should live in `CurerentUserController`. `refreshToken` must IMO live in the `Client` since it must be accessible all the time and it would be strange if `Client` has to create a controller to do something itself. `Client` shouldn't even know something like controllers exist.